planner/core/logical_plan_builder.go

```diff
@@ -4337,46 +4331,28 @@ func (b *PlanBuilder) buildUpdate(ctx context.Context, update *ast.UpdateStmt) (
 		tblID2table[id], _ = b.is.TableByID(id)
 	}
 	updt.TblColPosInfos, err = buildColumns2Handle(updt.OutputNames(), tblID2Handle, tblID2table, true)
-	if err == nil {
-		err = checkUpdateList(b.ctx, tblID2table, updt)
-	}
 	updt.PartitionedTable = b.partitionedTable
+	updt.tblID2Table = tblID2table
```

Comment:
`tblID2Table` is built twice, once in `PlanBuilder.buildUpdate` building from `tblID2Handle`, another is in `executorBuilder.buildUpdate` building from `TblColPosInfos`. And they are quite similar.

Can we remove the one in `Update` plan, if the only usage of `Update.tblID2Table` is for `CheckUpdateList`?

Reply:
@dyzsr great idea, but it seems executorBuilder.buildUpdate take care more about partitions?

how about add another issue to improve it? this's PR is release-block bugfix, introduce addition modification is high risk